google_guest_agent_packaging_diff_tool.py

```diff
@@ -1,46 +1,114 @@
-            relations = deb822.PkgRelation.parse_relations(ubuntu_control_fields[field_name])
-            print(relations)
-            for relation in relations:
-                for relation_item in relation:
-                    print(f'\tubuntu: {relation_item["name"]} {relation_item["version"]}')
-        else:
-            print(f'\tubuntu: None')
+@click.option('--agent-package', required=True,
+              type=click.Choice(["gce-compute-image-packages",
+                                        "google-compute-engine-oslogin",
+                                        "google-guest-agent",
+                                        "google-osconfig-agent"], case_sensitive=False),
+              help='The package to inspect')
+
+def main(agent_package: str) -> None:
+
+    url_list = archive_dict[agent_package]
+
+    ubuntu_req = requests.get(url_list[0])
+    if ubuntu_req.ok:
+        with open(UBUNTU_CONTROL_FILENAME, "w+") as fp:
+            fp.write(ubuntu_req.text)
+    else:
+        sys.exit(f"Error {ubuntu_req.status_code} - is {url_list[0]} still available?")
+
+    google_req = requests.get(url_list[1])
+    if google_req.ok:
+        with open(GOOGLE_CONTROL_FILENAME, "w+") as fp:
+            fp.write(google_req.text)
+    else:
+        sys.exit(f"Error {google_req.status_code} - is {url_list[1]} still available?")
+
+    ubuntu_control_fields = parse_control_file(UBUNTU_CONTROL_FILENAME)
+    upstream_control_fields = parse_control_file(GOOGLE_CONTROL_FILENAME)
+
+    # Compare the control fields
+    for field_name in [field_to_compare for field_to_compare in ubuntu_control_fields.keys() if field_to_compare in FIELDS_TO_COMPARE]:
+        with warnings.catch_warnings():
+            warnings.filterwarnings("error")
+            print(f"\n############## Comparing \"{field_name}\" ##############")
+            if field_name in upstream_control_fields.keys():
+                print(f"\"{field_name}\" field in Google upstream:\n {upstream_control_fields[field_name]}")
+                try:
+                    upstream_relations = deb822.PkgRelation.parse_relations(upstream_control_fields[field_name])
+                    pprint.pprint(upstream_relations, indent=4, sort_dicts=False)
+                    print("------------")
+                except Warning:
+                    pass
+            else:
+                print(f"\"{field_name}\" field not present in Google Upstream\n------------")
+            if field_name in ubuntu_control_fields.keys():
+                print(f"\"{field_name}\" field in ubuntu:\n {ubuntu_control_fields[field_name]}")
+                try:
+                    ubuntu_relations = deb822.PkgRelation.parse_relations(ubuntu_control_fields[field_name])
+                    pprint.pprint(ubuntu_relations, indent=4, sort_dicts=False)
+                except Warning:
+                    pass
+            else:
+                print(f"\"{field_name}\" field not present in ubuntu")
+
+    clean_up()
```

Comment:
Nit: What do you think about introducing `try...except` for everything after creating the files and having the `clean_up` call in a `finally` block?

Reply:
Happy to do that :smile: